ui/src/groups/GroupAdmin/GroupInfoEditor.tsx

```diff
@@ -59,14 +58,26 @@ export default function GroupInfoEditor({ title }: ViewProps) {
     [setDeleteField]
   );
 
-  const onDelete = useCallback(() => {
-    useGroupState.getState().delete(groupFlag);
-    navigate('/');
+  const onDelete = useCallback(async () => {
+    setDeleteStatus('loading');
+    try {
+      await useGroupState.getState().delete(groupFlag);
+      setDeleteStatus('success');
+      navigate('/');
```

Comment:
if we switch to a controlled dialog you could leave it up with status and then close it here